### DISCUSSION TITLE
LMI Solvers

### DISCUSSION THREAD

**Discussion by ayyobasa** [yalmip/YALMIP#962] [Optimization modelling] [answered]
Dear Johan

Hello

During working on a problem

I have found the following MI (matrix inequality) which is not symmetric 

I am thinking that the current solvers can solve it? (As far as I know, they must be symmetric i.e LMI problems)

if no how can I symmetrize it? 

I was going to use P+P^T /2 but I do not know how to apply it. 

( How must I symmetrize it without affecting the main problem)

![my lmi](https://user-images.githubusercontent.com/87689237/127746-6613dc58-b058-467b-95ff-6ecac75b64f3.PNG)

Thanks in advances

**Comment by johanlofberg** [comment, marked as answer]
A non-symmetric eigenvalue cannot be converted to a symmetric semi-definite cone constraint, and it cannot be solved as it is a much much harder problem.

However, I suspect you have derived the constraint erroneously. It looks like something coming from Lyapunov reasoning, and that should never lead to a non-symmetric eigenvalue constraint

**Reply by ayyobasa** [reply]
Thank you so much for your reply I need to check it again 

If again this result is gained then what is your suggestion to solve it?

I was thinking of writing a numerical alg but this is far from my current study.

Again so many thanks.

**Reply by johanlofberg** [reply]
I can only assume that P is a Lyapunov matrix, and thus the whole machinery is about decay of a quadratic function and by construction it will be symmetric. The fact that you have C'PC in there strengthens that suspicion, and the fact that you only have A^TP but not PA smells like an error too.

**Reply by ayyobasa** [reply]
Hello Johan 

Thanks a lot for your reply I have rechecked the computations and you are completely right there were some notes that 

I have neglected.

Did I understand right that P^T A must be obtained in the results?

However, I have obtained A diagonal matrix the term A^TP is still involved in the matrix, without its transpose P^T A/

Again I really appreciate you.

**Reply by johanlofberg** [reply]
Semidefinite programming based systems theory is based on the functional x'*Px and with dx/dt= Ax you have the derivative x'(PA+A'P)x from which you start extracting the LMI. Everything else is just variations on the same thing

**Reply by ayyobasa** [reply]
= x'^TPx=(Ax)^TPx=x^T(A^TP)x  Do I miss something?

or 
 = 2x'^TPx=2(Ax)^TPx=x^T(2A^TP)x 

I could not obtain PA

**Reply by johanlofberg** [reply]
Derivative is (dxdt)'Px + x'P(dxdt) = (Ax)'Px + x'P(Ax) = x'(A'P+PA)x

**Reply by ayyobasa** [reply]
Yes it is true I did wrong thank u very much

**Reply by ayyobasa** [reply]
![2elmimy](https://user-images.githubusercontent.com/87689237/128286469-66911225-ce97-477e-a10d-787fadcf51e9.PNG)

![1eeeeelmi](https://user-images.githubusercontent.com/87689237/128286436-788a1b04-852a-45a3-b6a4-69f68af30f0c.PNG)
Dear Johan 

For about er  of an LMI I have used 
it will lead to a   symmetric LMI but I'm not sure that if would it be solvable again 

`\Lambda=
\begin{bmatrix}
\Xi & 0 & 0 & 0& 0 & 0 \\
0 &2P\Pi_1^{-1}P^T & 0 & 0& 0& 0  \\
0 & 0 & B^TP\Pi_2^{-1}P^TB & 0 & 0 & 0\\
0 & 0 & 0 & 2P\Pi_3^{-1}P^T& 0& 0   \\
0 & 0 & 0 & 0 & B^T\Pi_4^{-1}B& 0 \\
0 & 0 & 0 & 0 & 0&(2P-C^TS)\Pi_5^{-1} \\
\end{bmatrix}
< 0`
in which

`\Xi=&A^TP+P^TA+(\Pi_1+\Pi_2+\Pi_3+\Pi_4+\Pi_5)\frac{\epsilon}{2}-\rho P-C^TQC\\`

If not are there other upper bounds that help me in finding better results?

**Reply by johanlofberg** [reply]
I have no idea what you are talking about with upper bounds etc. That matrix is not an LMI as the last block is not symmetric generally, and it is nonlinear in P (and if Pi is positive definite, most blocks are trivially infeasible, and trivially satisfied if Pi is negative definite

**Reply by ayyobasa** [reply]
Thank you so much
